### ISSUE TITLE
Type.GetGenericArguments() doesn't seem to work with v16-beta5

### ISSUE THREAD

**Issue by Zaid-Ajaj** [bridgedotnet/Bridge#2986]
Unable to extract generic type arguments from a generic type with v16-beta5 while it worked correctly with v16-beta4. I am assuming reflection is enabled by default at deck.net since I am able to log the method name in the code sample below. 

### Steps To Reproduce (v16-beta5)

[Deck Sample](https://deck.net/04a46660a63606202a908acc58146075)

```c#
public interface ISomeInterface
{
    Task<string> GetString();
}

public class Program
{
    public static void Main()
    {
        var type = typeof(ISomeInterface);
        var methods = type.GetMethods();
        var firstMethod = methods[0];
        Console.WriteLine(firstMethod.Name);
        var genericArgs = firstMethod.ReturnType.GetGenericArguments();
        Console.WriteLine(genericArgs.Length);
    }
}
```

### Expected Result

```js
GetString
1
```

### Actual Result

```js
GetString
0
```

**Comment by vladsch** [comment]
It is not a bug, Task<T> type is marked as IgnoreGeneric (javascript representation uses type without generic arguments). If you replace Task<string> by another type then it works
https://deck.net/015a319d561bc7409b34008ab3e4180c

**Comment by Zaid-Ajaj** [comment]
I am working on a library where I need to extract the type information specifically from a `Task<T>` 😞  is there a workaround? Is this temporary?

**Comment by LeonidVeriga** [comment]
@Zaid-Ajaj, we fixed this issue (no workaround); the fix will be included into **16.1**.

**Comment by Zaid-Ajaj** [comment]
@LeonidVeriga This is awesome and will help a lot! thank you :smile:
